packages/server/tests/acceptance/htsPrecompile.spec.ts

```diff
@@ -517,55 +511,104 @@ describe('HTS Precompile Acceptance Tests', async function () {
   });
 
   describe('HTS Precompile KYC Tests', async function() {
+    async function checkKyc(contractOwner, tokenAddress, accountAddress, expectedValue: boolean) {
+      const tx = await contractOwner.isKycPublic(tokenAddress, accountAddress, { gasLimit: 1_000_000 });
+      const responseCodeIsKyc = (await tx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode;
+      expect(responseCodeIsKyc).to.equal(TX_SUCCESS_CODE);
+  
+      const isKycGranted = (await tx.wait()).events.filter(e => e.event === 'KycGranted')[0].args.kycGranted;
+      expect(isKycGranted).to.equal(expectedValue);
+    };
+
     async function checkTokenDefaultKYCStatus(contractOwner, tokenAddress, expectedValue: boolean) {
       const txTokenDefaultStatus = await contractOwner.getTokenDefaultKycStatusPublic(tokenAddress, { gasLimit: 1_000_000 });
       const responseCodeTokenDefaultStatus = (await txTokenDefaultStatus.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode;
       const defaultTokenKYCStatus = (await txTokenDefaultStatus.wait()).events.filter(e => e.event === 'TokenDefaultKycStatus')[0].args.defaultKycStatus;
       expect(responseCodeTokenDefaultStatus).to.equal(TX_SUCCESS_CODE);
       expect(defaultTokenKYCStatus).to.equal(expectedValue);
-    }
+    };
 
     it('should be able to get default KYC status for fungible token', async function() {
-      await checkTokenDefaultKYCStatus(baseHTSContractOwner, HTSTokenContractAddress, true);
+      await checkTokenDefaultKYCStatus(baseHTSContractOwner, HTSTokenContractAddress, false);
     });
 
     it('should be able to get default KYC status for non fungible token', async function() {
-      await checkTokenDefaultKYCStatus(baseHTSContractOwner, NftHTSTokenContractAddress, true);
+      await checkTokenDefaultKYCStatus(baseHTSContractOwner, NftHTSTokenContractAddress, false);
     });
-  });
-
-  it('should be able to get a custom token fees', async function() {
-    const baseHTSContract = new ethers.Contract(BaseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
-
-    const tx = (await baseHTSContract.getTokenCustomFeesPublic(HTSTokenWithCustomFeesContractAddress));
-    const { fixedFees, fractionalFees } = (await tx.wait()).events.filter(e => e.event === 'TokenCustomFees')[0].args;
 
-    expect(fixedFees[0].amount).to.equal(1);
-    expect(fixedFees[0].tokenId).to.equal(HTSTokenContractAddress);
-    expect(fixedFees[0].useHbarsForPayment).to.equal(false);
-    expect(fixedFees[0].useCurrentTokenForPayment).to.equal(false);
-
-    expect(fractionalFees[0].numerator).to.equal(4);
-    expect(fractionalFees[0].denominator).to.equal(5);
-    expect(fractionalFees[0].minimumAmount).to.equal(10);
-    expect(fractionalFees[0].maximumAmount).to.equal(30);
-    expect(fractionalFees[0].netOfTransfers).to.equal(false);
+    it('should be able to grant Kyc', async function() {
```

Comment:
Are these set of KYC tests assuming they run sequentially?
Probably not good practice.
You could combine this into one test where you
- verify kyc is false to start
- grant kyc
- verify it's true
- verify token can now be sent
- revoke kyc
- verify it's false

At the acceptance test level this is acceptable :)

Reply:
Done. This flow was transferred to one test.